pgmpy/estimators/MLE.py

```diff
@@ -42,22 +43,26 @@ class MaximumLikelihoodEstimator(ParameterEstimator):
     """
 
     def __init__(self, model, data, **kwargs):
-        if not isinstance(model, BayesianNetwork) and not isinstance(
-            model, JunctionTree
-        ):
+        if not isinstance(model, (BayesianNetwork, JunctionTree, DAG)):
```

Comment:
The checks here should be in the following order:
```
if not isinstance(model, (DAG, BayesianNetwork, JunctionTree)):
       raise error

if isinstance(model, (DAG, BayesianNetwork)):
      if len(model.latents) > 0:
           Error MLE doesn't support latent variables
      elif len(set(data.columns) - set(model.nodes())) != 0:
            Error data not available for some nodes

if isinstance(model, JunctionTree):
      if  len(set(data.columns) - set(itertools.chain(*model.nodes()))) != 0:
              Error data not available for some nodes
```

Main points: 
1. DAG and BayesianNetworks are almost the same.
2. For Junction Tree, the node names are tuples of variables.
3. MLE doesn't support any latent variables.